core/dbt/events/types.py

```diff
@@ -38,908 +33,918 @@
 # | E    | DB adapter          |
 # | I    | Project parsing     |
 # | M    | Deps generation     |
-# | Q    | Node processing     |
+# | Q    | Node execution     |
 # | W    | Node testing        |
-# | Y    | Post processing     |
 # | Z    | Misc                |
 # | T    | Test only           |
 #
 # The basic idea is that event codes roughly translate to the natural order of running a dbt task
 
-# TODO: remove ingore when this is fixed:
-# https://github.com/python/mypy/issues/5374
 
+def format_adapter_message(name, base_msg, args) -> str:
+    # only apply formatting if there are arguments to format.
+    # avoids issues like "dict: {k: v}".format() which results in `KeyError 'k'`
+    msg = base_msg if len(args) == 0 else base_msg.format(*args)
+    return f"{name} adapter: {msg}"
 
-@dataclass  # type: ignore
-class AdapterEventBase(EventSerialization, Event):
-    name: str
-    base_msg: str
-    args: Tuple[Any, ...]
 
-    # instead of having this inherit from one of the level classes
-    def level_tag(self) -> str:
-        raise Exception("level_tag should never be called on AdapterEventBase")
-
-    def message(self) -> str:
-        # this class shouldn't be createable, but we can't make it an ABC because of a mypy bug
-        if type(self).__name__ == "AdapterEventBase":
-            raise Exception(
-                "attempted to create a message for AdapterEventBase which cannot be created"
-            )
-
-        # only apply formatting if there are arguments to format.
-        # avoids issues like "dict: {k: v}".format() which results in `KeyError 'k'`
-        msg = self.base_msg if len(self.args) == 0 else self.base_msg.format(*self.args)
-        return f"{self.name} adapter: {msg}"
+# =======================================================
+# A - Pre-project loading
+# =======================================================
 
 
 @dataclass
-class AdapterEventDebug(DebugLevel, AdapterEventBase, ShowException):
-    code: str = "E001"
-
+class MainReportVersion(InfoLevel, pt.MainReportVersion):  # noqa
+    def code(self):
+        return "A001"
 
-@dataclass
-class AdapterEventInfo(InfoLevel, AdapterEventBase, ShowException):
-    code: str = "E002"
+    def message(self):
+        return f"Running with dbt{self.version}"
```

Comment:
Should this include the log version as well?  You're passing it in in `main.py`.

Reply:
That would change the terminal output, which I don't think is necessary for the log format, just wanted to put it in a structured entry someplace.  We can discuss what to do about the log format.